thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/context/AnomalyDetectionContext.java

```diff
@@ -16,16 +18,16 @@
   private TimeSeriesKey timeSeriesKey;
   private long bucketSizeInMS; // the bucket size, gap between timestamps, in millisecond
 
-  private TimeSeries current;
-  private List<TimeSeries> baselines;
+  private Map<String, TimeSeries> current = new HashMap<>();
+  private Map<String, List<TimeSeries>> baselines = new HashMap<>();
 
   //TODO: Add DAO for accessing historical anomalies or scaling factor
 
   // The followings are intermediate results and are appended during anomaly detection
-  private TimeSeries transformedCurrent;
-  private List<TimeSeries> transformedBaselines;
+  private Map<String, TimeSeries> transformedCurrent = new HashMap<>();
+  private Map<String, List<TimeSeries>> transformedBaselines = new HashMap<>();
```

Comment:
Is it possible to treat baselines as named <string, timeseries>? your current choice is a direct extension from previous case where you have current  + baselines (list). But if you have current already as a list, there isnot strong reason you treat them differently. you just need the hash-name to impose the relationship between current and baselines, this relationship can be achieved with a real tag (or how you construct the transformer).